hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RegionScanner.java

```diff
@@ -115,4 +116,12 @@
    */
   boolean nextRaw(List<Cell> result, ScannerContext scannerContext)
       throws IOException;
+
+  default byte[] getStartRow() {
```

Comment:
We should not be exposing these in this interface. This interface is exposed to CPs possibly allowing to pollute the keys.
We can add a method which can check whether a full region scan or not. The RegionScannerImpl having ref to Scan instance as well as Region so it can check whether full region scan or not and return boolean

Reply:
thx for the suggestion - fixed, plz check